lib/rspec/matchers/built_in/has.rb

```diff
@@ -7,17 +7,10 @@ module BuiltIn
       class DynamicPredicate < BaseMatcher
         include BeHelpers
 
-        if RSpec::Support::RubyFeatures.kw_args_supported?
-          binding.eval(<<-CODE, __FILE__, __LINE__)
-          def initialize(method_name, *args, **kwargs, &block)
-            @method_name, @args, @kwargs, @block = method_name, args, kwargs, block
-          end
-          CODE
-        else
-          def initialize(method_name, *args, &block)
-            @method_name, @args, @block = method_name, args, block
-          end
+        def initialize(method_name, *args, &block)
+          @method_name, @args, @block = method_name, args, block
         end
+        ruby2_keywords :initialize if respond_to?(:ruby2_keywords, true)
```

Comment:
The key bit I was missing before was accidentally not including private methods when checking for ruby2_keywords I think, there is some thinking to do around how we handle this in Ruby 3 I suspect but for now this passes both the existing tests and the new one this PR adds.

Reply:
Yes, that is a tricky bit (not forgetting the `, true`), it also bit me once.
Ruby 3 still has `ruby2_keywords`, so it will work fine.
On Ruby 3, `*args, **kwargs`-delegation like above works correctly, but it's probably not worth maintaining 2 versions of the method while `ruby2_keywords` exists (the current plan seems to remove it after Ruby 2 EOL).